velox/core/QueryCtx.h

```diff
@@ -170,8 +169,10 @@ class ExecCtx {
   /// Prefer using LocalSelectivityVector which takes care of returning the
   /// vector to the pool on destruction.
   std::unique_ptr<SelectivityVector> getSelectivityVector(int32_t size) {
-    if (!isExpressionEvaluationCacheEnabled_ ||
-        selectivityVectorPool_.empty()) {
+    if (!exprEvalCacheEnabled_) {
```

Comment:
How about
```
if (!exprEvalCacheEnabled_) {
   VELOX_CHECK(selectivityVectorPool_.empty());
   return std::make_unique<SelectivityVector>(size);
}

```

Reply:
It's possible that selectivityVectorPool_.empty() is true even if exprEvalCacheEnabled_ is true too. So we need to handle this case.

Reply:
I see.
```
VELOX_CHECK(exprEvalCacheEnabled_ || selectivityVectorPool_.empty());
```